modules/sdk-coin-opeth/test/unit/transferBuilder.ts

```diff
@@ -0,0 +1,187 @@
+      const builder = new TransferBuilder()
+        .coin('topeth:terc18dp')
+        .expirationTime(1590078260)
+        .amount(amount)
+        .to(toAddress)
+        .contractSequenceId(2)
+        .key(key);
+      const result = builder.signAndBuild();
+      should.equal(result, testData.SEND_OPETH_TERC_DATA);
+    });
+
+    it('erc20 transfer should succeed', async () => {
+      const builder = new TransferBuilder()
+        .coin('topeth:terc18dp')
+        .expirationTime(1590078260)
+        .amount(amount)
+        .to(toAddress)
+        .contractSequenceId(0)
+        .key(key);
+      const result = builder.signAndBuild();
+      should.equal(result, testData.SEND_TOKEN_SEQUENCE_ZERO_DATA);
+    });
+
+    it('erc20 transfer with amount 0 should succeed', async () => {
+      const builder = new TransferBuilder()
+        .coin('topeth:terc18dp')
+        .expirationTime(1590078260)
+        .amount('0')
+        .to(toAddress)
+        .contractSequenceId(2)
+        .key(key);
+      const result = builder.signAndBuild();
+      should.equal(result, testData.SEND_TOKEN_AMOUNT_ZERO_DATA);
+    });
+
+    it('should build without a signature set', () => {
+      const builder = new TransferBuilder()
+        .expirationTime(1590078260)
+        .amount(amount)
+        .to(toAddress)
+        .contractSequenceId(2)
+        .data('0x');
+      const result = builder.signAndBuild();
+      should.equal(result, testData.SEND_FUNDS_NO_KEY_DATA);
+    });
+
+    it('should build from a non signed serialized data', () => {
+      const builder = new TransferBuilder(testData.SEND_FUNDS_NO_KEY_DATA);
+      builder.coin('topeth').key(key);
+      const result = builder.signAndBuild();
+      should.equal(result, testData.SEND_FUNDS_DATA);
+    });
+  });
+
+  describe('should fail', () => {
+    it('should fail if a coin does not exists in @bitgo/statics', () => {
+      should(() => {
+        new TransferBuilder().coin('inexistentcoin');
+      }).throw();
+    });
+
+    it('should fail with an invalid key', () => {
+      const builder = new TransferBuilder()
+        .coin('topeth')
+        .expirationTime(1590078260)
+        .amount(amount)
+        .to(toAddress)
+        .contractSequenceId(2)
+        .key('invalidkey');
+      should(() => {
+        builder.signAndBuild();
+      }).throw('private key length is invalid');
+    });
+
+    it('should fail with an invalid sequence id', () => {
+      should(() => {
+        new TransferBuilder().contractSequenceId(-1);
+      }).throw('Invalid contract sequence id');
+    });
+
+    it('should fail with an invalid destination address', () => {
+      should(() => {
+        new TransferBuilder().to('invalidaddress');
+      }).throw('Invalid address');
+    });
+
+    it('should fail with an invalid amount: text value', () => {
+      should(() => {
+        new TransferBuilder().amount('invalidamount');
+      }).throw('Invalid amount');
```

Comment:
Same as arbeth comments, can we move to a common place?

Reply:
moved these tests to abstract-eth module